### ISSUE TITLE
Customer coco dataset in self-training

### ISSUE THREAD

**Issue by tianyufang1958** [facebookresearch/CutLER#23]
Thanks for the nice work. I have a question regarding the customer coco dataset used in self-training. For my coco data, I have instances_train.py and instances_val.py, and I registered two datasets for both train and val, but in the first step of self-training, --test-dataset only take the 'imagenet_train'. 

Does it mean Imagenet only use one json file for both train and validation? Or json file generation of self-training can only be applied to training data itself not val data. I am confused about it.

**Comment by frank-xwang** [comment]
Duplicate of https://github.com/facebookresearch/CutLER/issues/16. Please check https://github.com/facebookresearch/CutLER/issues/16 for more details on working with custom datasets.

About the self-training dataset, you can train CutLER on any dataset you specify. But you must let the model know which dataset/split to work on by changing the command accordingly.

**Comment by frank-xwang** [comment]
If you plan to use pseudo-masks for your validation dataset, then it is necessary to provide the path to the dataset that contains the validation split using the "--dataset-path" argument.

**Comment by tianyufang1958** [comment]
> If you plan to use pseudo-masks for your validation dataset, then it is necessary to provide the path to the dataset that contains the validation split using the "--dataset-path" argument.

@frank-xwang My understanding is firstly use the whole imaging dataset to generate the pseudo masks. After that the dataset can be splits into training and validation like 80% and 20% as the inputs of the phase 2 training. Could you please confirm if this is correct?

**Comment by frank-xwang** [comment]
No, for self-training, we still utilize 100% of the data. Our experimental setup is: using all ImageNet data as the training set and evaluates the model's performance on 11 different detection datasets to demonstrate zero-shot unsupervised learning.

**Comment by frank-xwang** [comment]
Closing it now, please feel free to reopen it if you have further questions.
